src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/HtmlString.cs

```diff
@@ -39,29 +37,26 @@ public static HtmlString Empty
 
         /// <summary>
         /// Writes the value in this instance of <see cref="HtmlString"/> to the target
-        /// <paramref name="targetWriter"/>.
+        /// <paramref name="writer"/>.
         /// </summary>
-        /// <param name="targetWriter">The <see cref="TextWriter"/> to write contents to.</param>
-        public void WriteTo(TextWriter targetWriter)
+        /// <param name="writer">The <see cref="TextWriter"/> to write contents to.</param>
+        /// <param name="encoder">The <see cref="IHtmlEncoder"/> with which the output must be encoded.</param>
+        public void WriteTo(TextWriter writer, IHtmlEncoder encoder)
         {
-            if (_writer != null)
+            var writerAsStringCollectionTextWriter = writer as StringCollectionTextWriter;
+            if (writerAsStringCollectionTextWriter != null)
             {
-                _writer.CopyTo(targetWriter);
+                writerAsStringCollectionTextWriter.Content.Append(_input);
             }
             else
             {
-                targetWriter.Write(_input);
+                writer.Write(_input);
```

Comment:
What does this actually accomplish? I would expect `writerAsStringCollectionTextWriter.Content.Append(_input)` and `writer.Write(_input)` to do the same thing